kythe/java/com/google/devtools/kythe/analyzers/java/KytheTreeScanner.java

```diff
@@ -1596,6 +1596,23 @@ private void loadImplicitAnnotationsFile() {
     }
   }
 
+  private void loadInlineMetadata() {
+    String fullPath = filePositions.getSourceFile().toUri().getPath();
+    try {
+      if (metadataFilePaths.contains(fullPath)) {
+        return;
+      }
+      FileObject file = Iterables.getOnlyElement(fileManager.getJavaFileObjects(fullPath), null);
+      if (file == null) {
+        logger.atWarning().log("Can't find metadata at %s", fullPath);
+        return;
+      }
+      loadAnnotationsFile(fullPath, file);
```

Comment:
Rather than having the InlineMetadataLoader treat the Java source file as a big text blob, I suspect it will be more robust to use this class's access to the per-line comments and the metadata comment prefix to pull out only the relevant text directly, e.g.
```
for (Comment comment : comments.values()) {
  if (comment.text.startsWith(commentPrefix)) {
    loadAnnotationsData(fullPath,
     // Or something similar.
     comment.text.substring(commentPrefix.length()).getBytes(StandardCharsets.UTF_8));
  }
}
```

Where `loadAnnotationsData()` is most of the current body of `loadAnnotationsFile()`, which becomes:
```
 try (InputStream stream = file.openInputStream()) {
  return loadAnnotationsData(fullPath,  ByteStreams.toByteArray(stream));
 } catch (IOException ex) {
   logger.atWarning().withCause(ex).log("Can't read metadata for %s", fullPath);
 }
```

Reply:
This is a great suggestion! I've updated the PR with an implementation that mostly follows this structure.